zerver/lib/actions.py

```diff
@@ -1030,6 +1031,21 @@ def get_recipient_info(recipient: Recipient,
             'is_muted',
         ).order_by('user_profile_id')
 
+        exists_subquery = subscription_rows.filter(user_profile_id=OuterRef('sender_id'), is_muted=True)
+        follower_rows = stream_topic.get_followers_user_ids_for_topic(recipient).annotate(
+            user_profile_id=F('sender_id'),
+            email_notifications=F('sender__enable_topic_follow_email_notifications'),
+            push_notifications=F('sender__enable_topic_follow_push_notifications'),
+            wildcard_mentions_notify=F('sender__enable_topic_follow_wildcard_mentions_notify'),
+            is_muted=Exists(exists_subquery),
+        ).values(
+            'user_profile_id',
+            'email_notifications',
+            'push_notifications',
+            'wildcard_mentions_notify',
+            'is_muted',
```

Comment:
We shouldn't need this `is_muted` column, right?  Topic mutes are checked via `user_ids_muting_topic`, and we have the data on who has muted the stream via the `subscription_rows` query.  Best to not ask the database to do that `exists_subquery`.  We want this operation to be very fast, since we're doing it for every stream message.

So, I just realized that the index I added makes `messages_for_topic` fast, but not necessarily `get_followers_user_ids_for_topic`; it'll still need to look through all those messages to find the senders.  We might need yet another index that's on `(recipient_id, upper(topic), sender_id)` to support this query.